server/utils/hooks.py

```
@@ -92,11 +104,16 @@ def get_hook(name: str):
         fn = _HOOK_REGISTRY[name]
 
-    # Wrap in safety net so a broken hook never crashes the caller
+    # Hooks that gate access must fail closed; all others fail open.
+    _fail_closed_hooks = {"verify_entitlement"}
+
     def _safe_hook(*args, **kwargs):
         try:
             return fn(*args, **kwargs)
         except Exception as e:
+            if name in _fail_closed_hooks:
+                logger.error("Hook '%s' raised: %s — failing closed", name, e)
```

Ruff (0.15.17)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)